security/advancedtls/crl.go

```diff
@@ -0,0 +1,499 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package advancedtls
+
+import (
+	"bytes"
+	"crypto/sha1"
```

Comment:
FYI this package cannot be imported. `sha1` isn't an allowed package.
See error here:
https://fusion2.corp.google.com/invocations/7a79015c-6b2d-42c3-bb27-e6e88253cf38/targets/%2F%2Fthird_party%2Fgolang%2Fgrpc%2Fsecurity%2Fadvancedtls:advancedtls;config=a324982c22549be31b51b570f54afc4b6647c91d4c7c2ea021de868afff2a28b/log

All advancedtls code change will be ignored during import until this is fixed (I checked that there's currently no existing user of this in google3).

Reply:
Thanks for raising this issue Menghan!
It seems the problem can be easily fixed by replace `sha1` with `sha256`. I will apply a quick fix for this.
@mattjo in case I miss anything

Reply:
We cannot use sha256 here because it reimplements X509_NAME_hash from openssl function that uses sha1. The good news is that we don't actually care about hash collisions here and the use was approved in http://cl/354400609. Can we add the allowlist as a transformation on import into google3?

Reply:
@menghanl I can help with importing this change. I will ping you offline for more information.